beeline/src/test/org/apache/hive/beeline/TestCommands.java

```diff
@@ -30,21 +30,35 @@
+    assertEquals("show tables;", removeComments("show tables;"));
+    assertEquals("show tables;", removeComments("show tables; --comments"));
+    assertEquals("show tables;", removeComments("show tables; -------comments"));
+    assertEquals("show tables;", removeComments("show tables; -------comments;one;two;three;;;;"));
+    assertEquals("show", removeComments("show-- tables; -------comments"));
+    assertEquals("show", removeComments("show --tables; -------comments"));
+    assertEquals("s", removeComments("s--how --tables; -------comments"));
+    assertEquals("", removeComments("-- show tables; -------comments"));
 
-    assertEquals("\"show tables\"", removeComments("\"show tables\" --comments",escape));
-    assertEquals("\"show --comments tables\"", removeComments("\"show --comments tables\" --comments",escape));
-    assertEquals("\"'show --comments' tables\"", removeComments("\"'show --comments' tables\" --comments",escape));
-    assertEquals("'show --comments tables'", removeComments("'show --comments tables' --comments",escape));
-    assertEquals("'\"show --comments tables\"'", removeComments("'\"show --comments tables\"' --comments",escape));
+    assertEquals("\"show tables\"", removeComments("\"show tables\" --comments"));
+    assertEquals("\"show --comments tables\"", removeComments("\"show --comments tables\" --comments"));
+    assertEquals("\"'show --comments' tables\"", removeComments("\"'show --comments' tables\" --comments"));
+    assertEquals("'show --comments tables'", removeComments("'show --comments tables' --comments"));
+    assertEquals("'\"show --comments tables\"'", removeComments("'\"show --comments tables\"' --comments"));
+
+    assertEquals("show tables;", removeComments("--comments\nshow tables;"));
+    assertEquals("show tables;", removeComments("--comments\nshow tables; --comments"));
+    assertEquals("show tables;", removeComments("--comments\nshow tables; -------comments"));
+    assertEquals("show tables;", removeComments("--comments\nshow tables; -------comments;one;two;three;;;;"));
+    assertEquals("show", removeComments("--comments\nshow-- tables; -------comments"));
+    assertEquals("show", removeComments("--comments\nshow --tables; -------comments"));
+    assertEquals("s", removeComments("--comments\ns--how --tables; -------comments"));
+    assertEquals("", removeComments("--comments\n-- show tables; -------comments"));
+
+    assertEquals("\"show tables\"", removeComments("--comments\n\"show tables\" --comments"));
+    assertEquals("\"show --comments tables\"", removeComments("--comments\n\"show --comments tables\" --comments"));
+    assertEquals("\"'show --comments' tables\"", removeComments("--comments\n\"'show --comments' tables\" --comments"));
+    assertEquals("'show --comments tables'", removeComments("--comments\n'show --comments tables' --comments"));
+    assertEquals("'\"show --comments tables\"'", removeComments("--comments\n'\"show --comments tables\"' --comments"));
```

Comment:
Could you please add a test scenario where a multiline query strings has comments in between fragments of query? just as an example
"select col1,
--partitioned year column
year,
--partitioned month column
month,
--partitioned date column
date
from test_table
where
--for a particular user
username = 'foo';"

should return something equivalent to
"select col1, year, month, date from test_table where username = 'foo';"

Reply:
Thanks for having a look! I added your example. It contains both separate comment lines and inline comment.